manager/controlapi/secret.go

```diff
@@ -245,9 +242,11 @@ func validateSecretSpec(spec *api.SecretSpec) error {
 	if err := validateConfigOrSecretAnnotations(spec.Annotations); err != nil {
 		return err
 	}
-
-	if len(spec.Data) >= MaxSecretSize || len(spec.Data) < 1 {
-		return grpc.Errorf(codes.InvalidArgument, "secret data must be larger than 0 and less than %d bytes", MaxSecretSize)
+	if spec.Driver.Name != "" {
```

Comment:
I assume the built-in backend must have a name, or is likely to have a name at some point just to be explicit... maybe this check is not sufficient?